src/common/libflux/attr.h

```diff
@@ -72,6 +72,20 @@ const char *flux_get_hostbyrank (flux_t *h, uint32_t rank);
  */
 int flux_get_rankbyhost (flux_t *h, const char *host);
 
+typedef struct {
+    char text[128];
+} flux_lookup_error_t;
+
```

Comment:
Not sure if this is a great idea but I'll suggest it anyway - should we just call this `flux_error_t` and define it in <libflux/types.h> along with `flux_free_f`?  Then maybe we could stop proliferating similar typedefs like
```c
typedef struct {
    char filename[80];  // if unused, will be set to empty string
    int lineno;         // if unused, will be set to -1
    char errbuf[160];
} flux_conf_error_t;

typedef struct {
    char text[128];
} rlist_error_t;

typedef struct jobtap_error {
    char text [128];
} jobtap_error_t;

typedef struct {
    char text[160];
} job_list_error_t;

typedef struct {
    char text[160];
} flux_jobspec1_error_t;
```
Could also maybe define the useful `errprintf()` function.

Reply:
I had a similar thought, but didn't go there here (esp when it was tied up in another pr).

In general a good idea I think, I'll give it a try here and see how it goes.

Reply:
Oh, I remember the other thought I was having -- this will break existing users. Maybe better to do now than later though, but I'm not sure of the exact fallout...

Reply:
Maybe just name this with the general name and we can convert other APIs later?

Reply:
Of course I let myself off the hook when working on other PRs that did similar things, so it's a bit rude of me to call it out here :-)

Reply:
I could add the general name, and all existing cases could be typedef'd except for `flux_conf_error_t` which uses `errbuf` instead of `text` for the error text.